semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -458,10 +461,15 @@ let rec m_name a b =
                };
              _;
            } as infob) ) ) -> (
+      let is_resolved =
+        match resolved with
+        | B.ResolvedName _ -> true
```

Comment:
Should make this more general to cover `ImportedEntity` too. I think this issue is related to #4185, and I remember seeing other issues of this kind.